ruby/bob/bob.rb

```diff
@@ -0,0 +1,50 @@
+class Bob
+
+  def hey(remark)
+    if blank?(remark)
+      "Fine. Be that way!"
+    elsif other?(remark)
+      "Whatever."
+    elsif pure_question?(remark)
+      "Sure."
+    elsif yelling?(remark)
+      "Whoa, chill out!"
+    else
+      "Whatever."
+    end
+  end
+
```

Comment:
All of the methods below here aren't exposed publicly; you're only ever calling `hey` on an instance of the Bob class, so they can be marked as `private`.